dvc/external_repo.py

```diff
@@ -220,10 +341,6 @@ def _cached_clone(url, rev, for_write=False):
     logger.debug("erepo: making a copy of %s clone", url)
     copy_tree(clone_path, repo_path)
 
-    # Check out the specified revision
-    if rev is not None:
-        _git_checkout(repo_path, rev)
```

Comment:
If we don't checkout then structure of CLONES cache should be simplified.

Reply:
Another thing is this breaks current `for_write=True` use, which breaks how `dvcx` operates.

Reply:
@Suor this has been partially reverted, so that when `for_write=True` we will do git checkout (in a new repo kept separate from CLONES cache), and use WorkingTree like before. Will that work for dvcx?

Reply:
It should be, you may run dvcx tests to be sure.